experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -269,27 +261,34 @@ function Base.:(^)(x::WeylGroupElem, n::Int)
 end
 
 @doc raw"""
-    Base.:(<)(x::WeylGroupElem, y::WeylGroupElem)
+    <(x::WeylGroupElem, y::WeylGroupElem) -> Bool
 
-Returns whether `x` is smaller than `y` with respect to the Bruhat order.
+Returns whether `x` is smaller than `y` with respect to the Bruhat order,
+i.e., whether some (not necessarily connected) subexpression of a reduced
+decomposition of `y`, is a reduced decomposition of `x`.
 """
 function Base.:(<)(x::WeylGroupElem, y::WeylGroupElem)
   @req parent(x) === parent(y) "$x, $y must belong to the same Weyl group"
 
   if length(x) >= length(y)
     return false
+  elseif isone(x)
+    return true
   end
 
-  # x < y in the Bruhat order, iff some (not necessarily connected) subexpression
-  # of a reduced decomposition of y, is a reduced decomposition of x
-  j = length(x)
-  for i in length(y):-1:1
-    if word(y)[i] == word(x)[j]
-      j -= 1
-      if j == 0
+  tx = deepcopy(x)
+  for i in 1:length(y)
+    b, j, _ = explain_lmul(tx, y[i])
+    if b
```

Comment:
```suggestion
    if !b
```
I think this is now the wrong way around